src/close.rs

```diff
@@ -0,0 +1,99 @@
+use crate::{transaction::CloseTransaction, Channel};
+use bitcoin::{Address, Transaction};
+use ecdsa_fun::Signature;
+
+pub struct State0(Channel);
+pub struct State1 {
+    channel: Channel,
```

Comment:
If you don't need all the data in the `Channel` then it'd be better to extract it. This would allow someone _familiar_ with a protocol to recognise what step you're at.
Also, saves a bit of memory.

Reply:
Checkout the last commit. I'm not sure if this is "nicer" :)